symphony/bdk/core/datafeed/on_disk_datafeed_id_repository.py

```diff
@@ -0,0 +1,80 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from symphony.bdk.core.config.model.bdk_config import BdkConfig
+
+
+class DatafeedIdRepository(ABC):
+    """A repository interface for storing a datafeed id
+    By using the DatafeedLoopV1, the created datafeed id and agent base url should be persisted manually on the BDK side.
+    """
+
+    @abstractmethod
+    def write(self, datafeed_id: str, agent_base_path: str):
+        """Persists the created datafeed id into the storage.
+
+        :param datafeed_id: the datafeed id to be persisted
+        :param agent_base_path: the afent base path (i.e. scheme, host, port, context) to be persisted
+        """
+        pass
+
+    @abstractmethod
+    def read(self):
+        """Reads the persisted datafeed from storage.
+
+        :return the persisted datafeed id if present in storage, an empty string otherwise
+        """
+        pass
+
+    @abstractmethod
+    def read_agent_base_path(self):
+        """Reads the persisted agent base path from the storage.
+
+        :return: the persisted agent base path if present in storage, an empty string otherwise
+        """
+        pass
+
+
+class OnDiskDatafeedIdRepository(DatafeedIdRepository):
+    DATAFEED_ID_FILE = "datafeed.id"
+
+    def __init__(self, config: BdkConfig):
+        self.config = config
+
+    def write(self, datafeed_id, agent_base_path=""):
+        content = datafeed_id + "@" + agent_base_path
+        try:
+            self._get_datafeed_id_file().write_text(content, encoding='utf-8')
+        except ValueError:
```

Comment:
Why `ValueError`? What about I/O error (like file system permission issues)?

Reply:
This was for handling encoding errors but the content shouldn't be other than `utf-8` so I will remove it